### PR TITLE
fix: update resource with id

### PR DESCRIPTION
**Pull request by chengt-ing**

### CODE REVIEW

File: server/service/s3_object_presigner/s3_object_presigner.go [hunk 1]
```diff
@@ -45,9 +45,9 @@ func (p *S3ObjectPresigner) CheckAndPresign(ctx context.Context) {
 		}
 
 		if s3ObjectPayload.LastPresignedTime != nil {
-			// Skip if the presigned URL is still valid for the next 6 days.
-			// The default expiration time is 7 days.
-			if time.Now().Before(s3ObjectPayload.LastPresignedTime.AsTime().Add(6 * 24 * time.Hour)) {
+			// Skip if the presigned URL is still valid for the next 4 days.
+			// The expiration time is set to 5 days.
+			if time.Now().Before(s3ObjectPayload.LastPresignedTime.AsTime().Add(4 * 24 * time.Hour)) {
```
Comment (michaelien) [inline server/service/s3_object_presigner/s3_object_presigner.go:50]: Any context about this change?

File: server/service/s3_object_presigner/s3_object_presigner.go [hunk 2]
```diff
@@ -74,6 +74,7 @@ func (p *S3ObjectPresigner) CheckAndPresign(ctx context.Context) {
 		s3ObjectPayload.S3Config = s3Config
 		s3ObjectPayload.LastPresignedTime = timestamppb.New(time.Now())
 		if err := p.Store.UpdateResource(ctx, &store.UpdateResource{
+			ID:        resource.ID,
```
Comment (michaelien) [inline server/service/s3_object_presigner/s3_object_presigner.go:77]: Good catch!

**Review by michaelien — APPROVED** [review]

File: server/service/s3_object_presigner/s3_object_presigner.go (hunk 1 above)
Comment (chengt-ing) [reply server/service/s3_object_presigner/s3_object_presigner.go:50]: I saw that the commit https://github.com/usememos/memos/commit/cf7718f8dc98290a36c1210491130ad4b7b0ff63 changed the expiration time but did not update it in this file

**Comment by michaelien** [comment]
Thanks for your contribution. It will be cherrypicked to `v0.22.0`.

**Pull request merged**